### ISSUE TITLE
The `always` fix of `declaration-block-trailing-semicolon` wipes the run in front of a closing brace that a bodiless at-rule does not run to

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#684]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
	@content;
}
```

The `always` fix of `declaration-block-trailing-semicolon` writes the semicolon it adds behind a bodiless at-rule by moving that at-rule's trailing whitespace into the block's own raw, so that the semicolon lands in front of the run rather than behind it:

```js
bodilessAtRule.raws.between = beforeWhitespace
parent.raws.after = between.slice(beforeWhitespace.length)
```

That holds where the at-rule runs to the closing brace, which is the only way the parser hands one over with no semicolon of its own. Where a semicolon closed the at-rule at parse, `raws.between` ends at the at-rule's text and the run in front of the brace stands in the block's `raws.after` already — and the second line then overwrites it with the empty tail of `raws.between`.

The flag has to have been cleared in the same pass for the fix to run at all, which takes the rule under two namespaces with `never` listed first, both reading one plain CSS file.

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/scss/declaration-block-trailing-semicolon": "never",
		"@stylistic/declaration-block-trailing-semicolon": "always"
	}
}
```

`@stylistic/less/` in place of `@stylistic/scss/` reaches it the same way, and so does the reverse pair over `a { @content }`, which the `always` fix brings to this shape on its first run.

### What did you expect to happen?

The whitespace in front of the closing brace to stay where the file spells it. Neither option of the rule is about that run; the three `block-closing-brace-*-before` rules are.

### What actually happened?

The run goes, whatever it is spelled with, and a block written over three lines comes back with its closing brace on the at-rule's line:

| input | `--fix` |
| --- | --- |
| `a { @content; }` | `a { @content;}` |
| `a { @content;\t}` | `a { @content;}` |
| `a { @content;\n}` | `a { @content;}` |
| `a {\n\t@content;\n}\n` | `a {\n\t@content;}\n` |

A declaration in the same place is untouched: `a { color: pink; }` comes back as it went in.

### Notes

Found by the review of the pull request closing issue #630. Measured on `1949e23` and on that branch alike, so it is nothing that branch made. The write is unconditional, while the raw it takes the run from holds one only where the at-rule runs to the brace.
